File: Find a Pair/solve with js/solve-2.js
// Time complexity: O(nlogn)
// Space complexity: O(1) if you are allowed to modify the input, O(n) else


function findPair(arr,k){
	arr = arr.sort((a,b)=>a-b);
	left = 0,right =arr.length-1;
	while(left<right){
		if(left[i] = right[i] ==k) return true;
		else if(left[i] + right[i] <k) left++;
		else right--;
	}
	return false
}

File: Find a Pair/solve with c++/solve-2.cpp
bool findPair(vector<int> arr, int k){
  sort(arr.begin(), arr.end());
  int left = 0, right = arrLength-1;
  while(left < right){
    if(arr[left] + arr[right] == k) return true;
    else if(arr[left] + arr[right] < k) left++;
    else right--;
  }
  return false;
}

